src/main/java/org/roaringbitmap/RunContainer.java

```diff
@@ -942,6 +932,40 @@ public boolean equals(Object o) {
     return false;
   }
 
+  private boolean equals(RunContainer runContainer) {
+    if (runContainer.nbrruns != this.nbrruns) {
+      return false;
+    }
+    for (int i = 0; i < nbrruns; ++i) {
+      if (getValue(i) != runContainer.getValue(i)) {
+        return false;
+      }
+      if (getLength(i) != runContainer.getLength(i)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private boolean equals(ArrayContainer arrayContainer) {
+    int pos = 0;
+    for (short i = 0; i < nbrruns; ++i) {
+      short runStart = getValue(i);
+      short length = getLength(i);
+      if (pos + length >= arrayContainer.getCardinality()) {
+        return false;
```

Comment:
@richardstartin  The Go counterpart of this library has fuzzy testing: 

https://github.com/RoaringBitmap/roaring/blob/master/Makefile#L71

It is great to have as supplementary testing.

I don't think it replaces or even competes with unit testing.